src/python/pants/backend/python/target_types_rules.py

```diff
@@ -135,29 +142,124 @@ async def inject_pex_binary_entry_point_dependency(
 # -----------------------------------------------------------------------------------------------
 
 
+@rule(desc="Determining the entry points for a `python_distribution` target")
+async def resolve_python_distribution_entry_points(
+    request: ResolvePythonDistributionEntryPointsRequest,
+) -> ResolvedPythonDistributionEntryPoints:
+    field_value = request.entry_points_field.value
+    if field_value is None:
+        return ResolvedPythonDistributionEntryPoints(None)
+
+    address = request.entry_points_field.address
+    entry_points: dict[str, dict[str, EntryPoint]] = defaultdict(dict)
+
+    for section, values in field_value.items():
+        for name, ref in values.items():
+            if ref.startswith(":") or "/" in ref:
+                targets = await Get(Targets, UnparsedAddressInputs([ref], owning_address=address))
```

Comment:
We'll want to parallelize this via `MultiGet`. It's an antipattern to use `Get` inside a loop like this: https://www.pantsbuild.org/docs/rules-api-tips#tip-use-multiget-for-increased-concurrency

I'm not certain the best way to do that yet though, and it might be more clear after making below changes. One possible way is to factor out a helper rule which processes a single "section", then use `await MultiGet` where you have a `Get` for each section.